File: typescript/src/shapes/rectangle.ts
function Rectangle(width: number, height: number): Shape {
    let _width = width
    let _height = height

    function getWidth(): number {
        return _width
    }

    function getHeight(): number {
        return _height
    }

    return {
        computeArea: function (): number {
            return getWidth() * getHeight()
        }
    }
}

export { Rectangle }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Rectangle.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Rectangle implements Shape {
    private double height;
    private double width;
    
    public Rectangle(double height, double width){
        this.height = height;
        this.width = width;
    }

    public double getArea() {
        return getHeight() * getWidth();
    }

    private double getHeight() {
        return this.height;
    }

    private double getWidth() {
        return this.width;
    }
}
